google/cloud/bigquery/dbapi/_helpers.py

```diff
@@ -184,7 +189,11 @@ def bigquery_scalar_type(value):
     elif isinstance(value, numbers.Real):
         return "FLOAT64"
     elif isinstance(value, decimal.Decimal):
-        return "NUMERIC"
+        scalar_object = pyarrow.scalar(value)
+        if isinstance(scalar_object, pyarrow.Decimal128Scalar):
```

Comment:
Let's add a comment that we check for numeric before bignumeric in order to support pyarrow < 3.0